internal/config/controller.go

```diff
@@ -222,3 +236,151 @@ func (c *controller) getConfig(configName string) (*s2hv1beta1.Config, error) {
 
 	return config, nil
 }
+
+func (c *controller) ensureComponentChanged(teamName, namespace string) error {
+	comps, err := c.GetComponents(teamName)
+	if err != nil {
+		logger.Error(err, "cannot get components from configuration",
+			"team", teamName, "namespace", namespace)
+		return err
+	}
+
+	if err := c.deleteDesiredComponents(comps, namespace); err != nil {
+		return err
+	}
+
+	if err := c.deleteTeamDesiredComponents(comps, teamName); err != nil {
+		return err
+	}
+
+	if err := c.deleteQueues(comps, namespace); err != nil {
+		return err
+	}
+
+	if err := c.deleteStableComponents(comps, namespace); err != nil {
+		return err
+	}
```

Comment:
what if new components have been added? 
should we auto-update the desired component and add in the queue also?